### PR TITLE
Add unit tests for SAMLAssertionClaimsCallback

### PR DESCRIPTION
**Pull request by GDRDABARERA**
### Proposed changes in this pull request

[List all changes you want to add here. If you fixed an issue, please
add a reference to that issue as well.]

-

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallback.java [hunk 1]
```diff
@@ -95,8 +95,9 @@
     static {
         UserRealm realm;
         try {
-            realm = OAuthComponentServiceHolder.getInstance().getRealmService().getTenantUserRealm
-                    (MultitenantConstants.SUPER_TENANT_ID);
+            realm = OAuthComponentServiceHolder.getInstance()
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallback.java]: Formatting issue

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
+package org.wso2.carbon.identity.openidconnect;
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java]: Add license

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import org.wso2.carbon.user.api.RealmConfiguration;
+import org.wso2.carbon.user.api.UserRealm;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.service.RealmService;
+
+import static org.mockito.Mockito.mock;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.assertTrue;
+
+
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java]: Remove unnecessary new line.

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java (hunk 2 above)
Comment (GDRDABARERA) [reply components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java]: done

File: components/org.wso2.carbon.identity.oauth/src/test/resources/testng.xml [hunk 4]
```diff
@@ -17,9 +17,8 @@
   ~ under the License.
   -->
 
-<suite name="Suite1" verbose="1" >
-
-    <test name="Regression1">
+<suite name="Suite1"  >
```
Comment (dharshanaw) [inline components/org.wso2.carbon.identity.oauth/src/test/resources/testng.xml]: Add propper name for suites

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+import java.util.List;
+
+import static org.mockito.Mockito.mock;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Class which tests SAMLAssertionClaimsCallback.
+ */
+
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java]: Remove new line.

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+        when(userRealm.getUserStoreManager()).thenReturn(userStoreManager);
+        realmConfiguration = mock(RealmConfiguration.class);
+        when(userStoreManager.getRealmConfiguration()).thenReturn(realmConfiguration);
+        when(realmConfiguration.getUserStoreProperty(IdentityCoreConstants.MULTI_ATTRIBUTE_SEPARATOR)).
+                thenReturn(IdentityCoreConstants.MULTI_ATTRIBUTE_SEPARATOR_DEFAULT);
+
+        samlAssertionClaimsCallback = new SAMLAssertionClaimsCallback();
+    }
+
+    @Test(priority = 0)
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java]: Do we need priority?

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+        assertTrue(jwtClaimsSet.getAllClaims().isEmpty());
+    }
+
+    @Test(priority = 1)
+    public void testCustomClaimHandleWithAssertion() throws Exception {
+        jwtClaimsSet = mock(JWTClaimsSet.class);
+        requestMsgCtx = mock(OAuthTokenReqMessageContext.class);
+        Assertion assertion = mock(Assertion.class);
+        AttributeStatement statement = mock(AttributeStatement.class);
+        List<AttributeStatement> attributeStatementList = new ArrayList<>();
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java]: Segment the code with newlines where necessary to make it more readable.

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+        when(oAuthAuthzReqMessageContext.getApprovedScope()).thenReturn(approvedScopes);
+        when(oAuthAuthzReqMessageContext.getProperty(OAuthConstants.ACCESS_TOKEN))
+                .thenReturn(SAMPLE_ACCESS_TOKEN);
+        mockStatic(AuthorizationGrantCache.class);
+        authorizationGrantCache = mock(AuthorizationGrantCache.class);
+        AuthorizationGrantCacheEntry authorizationGrantCacheEntry = mock(AuthorizationGrantCacheEntry.class);
+        when(AuthorizationGrantCache.getInstance()).thenReturn(authorizationGrantCache);
+        when(authorizationGrantCache.getValueFromCache(any(AuthorizationGrantCacheKey.class))).
+                thenReturn(authorizationGrantCacheEntry);
+        OAuth2AuthorizeReqDTO oAuth2AuthorizeReqDTO = mock(OAuth2AuthorizeReqDTO.class);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java]: Can we add a new line here to increase readability?

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+
+    @Test
+    public void testHandleClaimsForOAuthAuthzReqMessageContext() throws Exception {
+        jwtClaimsSet = mock(JWTClaimsSet.class);
+        oAuthAuthzReqMessageContext = mock(OAuthAuthzReqMessageContext.class);
+        String[] approvedScopes = {"openid", "testScope1", "testScope2"};
+        when(oAuthAuthzReqMessageContext.getApprovedScope()).thenReturn(approvedScopes);
+        when(oAuthAuthzReqMessageContext.getProperty(OAuthConstants.ACCESS_TOKEN))
+                .thenReturn(SAMPLE_ACCESS_TOKEN);
+        mockStatic(AuthorizationGrantCache.class);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallbackTest.java]: Can we add a new line here to increase readability?

**Pull request merged**